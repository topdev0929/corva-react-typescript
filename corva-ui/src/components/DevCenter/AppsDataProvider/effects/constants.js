export const APPS_DATA_REQUESTS = {
  WELLS: 'WELLS',
  RIGS: 'RIGS',
  FRAC_FLEETS: 'FRAC_FLEETS',
  FRAC_FLEET_WELLS: 'FRAC_FLEET_WELLS',
  PAD_WELLS: 'PAD_WELLS',
};
