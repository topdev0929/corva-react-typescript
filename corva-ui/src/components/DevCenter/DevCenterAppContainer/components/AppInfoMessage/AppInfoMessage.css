.appInfoMessage {
  display: flex;
  flex-direction: column;
  padding: 12px 12px 30px 12px;
  height: 100%;
}

.content {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100%;
}

.message {
  color: #bdbdbd;
  font-size: 24px;
  text-align: center;
  line-height: 1.5;
}

.submessage {
  color: #bdbdbd;
  font-size: 14px;
  text-align: center;
  line-height: 1.2;
}

.infoIcon {
  color: #bdbdbd;
  width: 7em !important;
  height: 7em !important;
}
