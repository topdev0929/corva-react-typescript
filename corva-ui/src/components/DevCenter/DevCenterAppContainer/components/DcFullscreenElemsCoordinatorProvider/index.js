export {
  DcFullscreenElemsCoordinatorProvider,
  useDcFullscreenElemsCoordinator,
} from './DcFullscreenElemsCoordinatorProvider';
export {
  IsInsideDcFullscreenElemProvider,
  useIsInsideDcFullscreenElem,
} from './IsInsideDcFullscreenElemProvider';
