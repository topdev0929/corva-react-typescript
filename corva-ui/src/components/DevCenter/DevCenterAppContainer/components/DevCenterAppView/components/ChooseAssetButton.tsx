import { Button, makeStyles } from '@material-ui/core';

const useStyles = makeStyles({ chooseAssetButton: { margin: '8px 0', width: 150 } });

interface ChooseAssetButtonProps {
  toggleAppSettingsDialog: () => void;
}

export const PAGE_NAME = 'Choose_asset_button';

export const ChooseAssetButton = ({
  toggleAppSettingsDialog,
}: ChooseAssetButtonProps): JSX.Element => {
  const styles = useStyles();

  return (
    <Button
      className={styles.chooseAssetButton}
      color="primary"
      onClick={toggleAppSettingsDialog}
      variant="contained"
      data-testid={PAGE_NAME}
    >
      CHOOSE ASSET
    </Button>
  );
};
