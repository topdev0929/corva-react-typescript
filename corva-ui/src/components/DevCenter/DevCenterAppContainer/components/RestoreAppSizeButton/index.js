import PropTypes from 'prop-types';
import classNames from 'classnames';

import { Icon, IconButton } from '@material-ui/core';

import { isNativeDetected } from '~/utils/mobileDetect';

import styles from './RestoreAppSizeButton.css';

const PAGE_NAME = 'DevCenter_AppContainer';

function RestoreAppSizeButton({ setIsMaximized }) {
  return (
    <div
      className={classNames(
        styles.containerRestoreAction,
        isNativeDetected && styles.containerRestoreActionNative
      )}
    >
      <IconButton
        data-testid={`${PAGE_NAME}_restore`}
        className={styles.iconButton}
        onClick={() => {
          setIsMaximized(false);
        }}
      >
        <Icon
          className={
            isNativeDetected ? styles.containerRestoreIconNative : styles.containerRestoreIcon
          }
        >
          close
        </Icon>
      </IconButton>
    </div>
  );
}

RestoreAppSizeButton.propTypes = {
  setIsMaximized: PropTypes.func.isRequired,
};

export default RestoreAppSizeButton;
