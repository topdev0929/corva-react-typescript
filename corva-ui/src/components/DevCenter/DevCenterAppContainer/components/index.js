export { default as AppActions } from './AppActions';
export { default as AppSettingsDialog } from './AppSettingsDialog';
export { default as AppSettingsDialogV2 } from './AppSettingsDialogV2';
export { default as AnnotationsButton } from './AnnotationsButton';
export { default as NonPriorityMenus } from './NonPriorityMenus';
export { default as FullscreenButton } from './FullscreenButton';
export { default as AppInfoMessage } from './AppInfoMessage';
export { default as DevCenterAppView } from './DevCenterAppView';
export {
  DcFullscreenElemsCoordinatorProvider,
  useDcFullscreenElemsCoordinator,
} from './DcFullscreenElemsCoordinatorProvider';
export { IsInsideDcAppProvider, useIsInsideDcApp } from './IsInsideDcAppProvider';
