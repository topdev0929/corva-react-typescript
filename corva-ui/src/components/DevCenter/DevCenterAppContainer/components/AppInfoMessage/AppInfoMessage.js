import PropTypes from 'prop-types';
import InfoOutlinedIcon from '@material-ui/icons/InfoOutlined';

import { AppHeader } from '~/components';

import styles from './AppInfoMessage.css';

export const PAGE_NAME = 'AppInfoMessage';

function AppInfoMessage({ app, children, message, submessage }) {
  return (
    <div className={styles.appInfoMessage} data-testid={PAGE_NAME}>
      <AppHeader app={app} />
      <div className={styles.content}>
        {children}
        {!children && (
          <>
            <InfoOutlinedIcon size="large" className={styles.infoIcon} />
            <p className={styles.message}>{message}</p>
            {submessage && <p className={styles.submessage}>{submessage}</p>}
          </>
        )}
      </div>
    </div>
  );
}

AppInfoMessage.propTypes = {
  app: PropTypes.shape({}).isRequired,
  children: PropTypes.node,
  message: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
  submessage: PropTypes.string,
};

AppInfoMessage.defaultProps = { message: '', submessage: '', children: null };

export default AppInfoMessage;
