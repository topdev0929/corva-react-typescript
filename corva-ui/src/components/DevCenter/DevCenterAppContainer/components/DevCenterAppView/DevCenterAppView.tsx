import ErrorBoundary from '~/components/ErrorBoundary';
import LoadingIndicator from '~/components/LoadingIndicator';
import { SEGMENTS } from '~/constants/segment';
import EmptyAppView from '~/components/EmptyView/EmptyAppView';
import EmptyState from '~/components/EmptyState';
import { ChooseAssetButton } from './components/ChooseAssetButton';
import AppInfoMessage from '../AppInfoMessage';

const PAGE_NAME = 'DevCenterAppView';

export const getNoActiveWellMessage = (rigName: string): string => `${rigName} has no active well`;

function DevCenterAppView({
  AppComponent,
  app,
  appData,
  appProps,
  availability,
  isCLIApp,
  isLoading,
  isMultiRig,
  layoutEnvironment,
  openIntercom,
  segment,
  toggleAppSettingsDialog,
}) {
  if (!app.package)
    return (
      <AppInfoMessage app={app}>
        <EmptyState
          data-testid={`${PAGE_NAME}_EmptyState`}
          title={EmptyState.APP_MESSAGES.appPackageWasNotFound.title}
          subtitle={EmptyState.APP_MESSAGES.appPackageWasNotFound.subtitle}
          image={EmptyState.IMAGES.AppLoadingError}
        />
      </AppInfoMessage>
    );

  if (isLoading)
    return (
      <AppInfoMessage app={app}>
        <LoadingIndicator />
      </AppInfoMessage>
    );

  if (!(availability.hasProductSubscription && availability.isAppSubscribedForAsset)) {
    const {
      APP_MESSAGES: { appNotSubscribed, appNotSubscribedForAsset },
      IMAGES: { NoSubscription },
    } = EmptyState;
    const { title, subtitle } = !availability.hasProductSubscription
      ? appNotSubscribed
      : appNotSubscribedForAsset;

    return (
      <AppInfoMessage app={app}>
        <EmptyState
          data-testid={`${PAGE_NAME}_EmptyState`}
          title={title}
          subtitle={subtitle}
          image={NoSubscription}
        />
      </AppInfoMessage>
    );
  }

  if (!AppComponent)
    return (
      <AppInfoMessage app={app}>
        <EmptyState
          data-testid={`${PAGE_NAME}_EmptyState`}
          title={EmptyAppView.APP_MESSAGES.appComponentWasNotFound.title}
          image={EmptyState.IMAGES.AppLoadingError}
        />
      </AppInfoMessage>
    );

  if (!isMultiRig) {
    const actions = isCLIApp && (
      <ChooseAssetButton toggleAppSettingsDialog={toggleAppSettingsDialog} />
    );

    if (appData.rig && !appData.well) {
      return (
        <AppInfoMessage app={app}>
          <EmptyState
            data-testid={`${PAGE_NAME}_EmptyState`}
            title={getNoActiveWellMessage(appData.rig.name)}
            image={EmptyState.IMAGES.RigHasNoActiveWell}
            subtitle={actions}
          />
        </AppInfoMessage>
      );
    }

    if (
      layoutEnvironment?.type === 'general' &&
      segment === SEGMENTS.COMPLETION &&
      !appData.fracFleet
    ) {
      return (
        <AppInfoMessage app={app}>
          <EmptyState
            data-testid={`${PAGE_NAME}_EmptyState`}
            title={EmptyAppView.APP_MESSAGES.fracFleetWasNotFound.title}
            image={EmptyState.IMAGES.FracFleetHasNoActivePad}
            subtitle={actions}
          />
        </AppInfoMessage>
      );
    }

    if (
      layoutEnvironment?.type === 'general' &&
      segment === SEGMENTS.COMPLETION &&
      !appData.fracFleet?.current_pad_id &&
      !app.settings.padId
    ) {
      return (
        <AppInfoMessage app={app}>
          <EmptyState
            data-testid={`${PAGE_NAME}_EmptyState`}
            image={EmptyState.IMAGES.FracFleetHasNoActivePad}
            subtitle={actions}
            title={EmptyState.APP_MESSAGES.fracFleetHasNoPad.title}
          />
        </AppInfoMessage>
      );
    }

    if (
      layoutEnvironment?.type === 'general' &&
      segment === SEGMENTS.COMPLETION &&
      (!appData.wells || appData.wells.length < 1)
    ) {
      return (
        <AppInfoMessage app={app}>
          <EmptyState
            data-testid={`${PAGE_NAME}_EmptyState`}
            image={EmptyState.IMAGES.NoWellsSelected}
            subtitle={actions}
            title={EmptyState.APP_MESSAGES.padHasNoWells.title}
          />
        </AppInfoMessage>
      );
    }

    if (!appData.rig && !appData.well && !appData.settings && !appData.wells) {
      return (
        <AppInfoMessage app={app}>
          <EmptyState
            data-testid={`${PAGE_NAME}_EmptyState`}
            image={EmptyState.IMAGES.AssetHasNoData}
            subtitle={actions}
            title={EmptyState.APP_MESSAGES.noAssetData.title}
          />
        </AppInfoMessage>
      );
    }
  }
  return (
    <ErrorBoundary app={app} openIntercom={openIntercom}>
      <AppComponent app={app} {...app.settings} {...appData} {...appProps} />
    </ErrorBoundary>
  );
}

DevCenterAppView.defaultProps = {
  availability: {
    hasProductSubscription: true,
    isAppSubscribedForAsset: true,
  },
};

export default DevCenterAppView;
