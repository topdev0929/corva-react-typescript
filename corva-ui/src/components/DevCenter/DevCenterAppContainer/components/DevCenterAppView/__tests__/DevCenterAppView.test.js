import { render, screen, fireEvent, waitFor } from '@testing-library/react';

import { APP_MESSAGES } from '~/components/EmptyView/EmptyAppView/EmptyAppView';
import { SEGMENTS } from '~/constants/segment';

import DevCenterAppView, { getNoActiveWellMessage } from '../DevCenterAppView';
import { PAGE_NAME as CHOOSE_DATA_TEST_ID } from '../components/ChooseAssetButton';
import { PAGE_NAME as APP_INFO_MESSAGE_TEST_ID } from '../../AppInfoMessage/AppInfoMessage';

const TEST_IDS = {
  LOADING_INDICATOR: 'loadingIndicator',
};
const APP = {
  package: { manifest: { application: { name: 'My awesome app' } } },
  settings: {},
};
const withProps = (props = {}) => <DevCenterAppView app={APP} {...props} />;

jest.mock('~/components/LoadingIndicator', () => () => (
  <div data-testid={TEST_IDS.LOADING_INDICATOR} />
));

const testChooseAssetButton = props =>
  describe('ChooseAssetButton', () => {
    it('does not render ChooseAssetButton for non CLI app', () => {
      const { queryByTestId } = render(withProps(props));

      expect(queryByTestId(CHOOSE_DATA_TEST_ID)).toBeNull();
    });

    describe('CLI app', () => {
      it('renders ChooseAssetButton', () => {
        const { getByTestId } = render(withProps({ ...props, isCLIApp: true }));

        expect(getByTestId(CHOOSE_DATA_TEST_ID)).toBeInTheDocument();
      });

      it('calls toggleAppSettingsDialog onClick', () => {
        const toggleAppSettingsDialog = jest.fn();
        const { getByTestId } = render(
          withProps({ ...props, isCLIApp: true, toggleAppSettingsDialog })
        );

        fireEvent.click(getByTestId(CHOOSE_DATA_TEST_ID));
        expect(toggleAppSettingsDialog).toHaveBeenCalled();
      });
    });
  });

describe('DevCenterAppView', () => {
  it('should render DevCenterAppView without errors', () => {
    const { container } = render(withProps());

    expect(container).toBeInTheDocument();
  });

  it(`renders ${APP_MESSAGES.appPackageWasNotFound.title} error if app has not package`, () => {
    const { getByText } = render(withProps({ app: {} }));

    expect(getByText(APP_MESSAGES.appPackageWasNotFound.title)).toBeInTheDocument();
    expect(getByText(APP_MESSAGES.appPackageWasNotFound.subtitle)).toBeInTheDocument();
  });

  it('renders loading indicator if isLoading', () => {
    const { getByTestId } = render(withProps({ isLoading: true }));

    expect(getByTestId(TEST_IDS.LOADING_INDICATOR)).toBeInTheDocument();
  });

  it(`renders ${APP_MESSAGES.appNotSubscribed.title} if app does not have PRODUCT subscription `, () => {
    const { getByText } = render(withProps({ availability: { hasProductSubscription: false } }));

    expect(getByText(APP_MESSAGES.appNotSubscribed.title)).toBeInTheDocument();
    expect(getByText(APP_MESSAGES.appNotSubscribed.subtitle)).toBeInTheDocument();
  });

  it(`renders ${APP_MESSAGES.appNotSubscribedForAsset.title} if app does not have ASSET subscription `, () => {
    const { getByText } = render(
      withProps({ availability: { hasProductSubscription: true, isAppSubscribedForAsset: false } })
    );

    expect(getByText(APP_MESSAGES.appNotSubscribedForAsset.title)).toBeInTheDocument();
    expect(getByText(APP_MESSAGES.appNotSubscribedForAsset.subtitle)).toBeInTheDocument();
  });

  it(`renders ${APP_MESSAGES.appComponentWasNotFound.title} if there is no AppComponent`, () => {
    const { getByText } = render(
      withProps({ availability: { hasProductSubscription: true, isAppSubscribedForAsset: true } })
    );

    expect(getByText(APP_MESSAGES.appComponentWasNotFound.title)).toBeInTheDocument();
  });

  describe('non MultiRigApp', () => {
    describe('drilling app', () => {
      const RIG_NAME = 'My awesome rig';
      const drillingAppProps = {
        availability: { hasProductSubscription: true, isAppSubscribedForAsset: true },
        AppComponent: {},
        appData: { rig: { name: RIG_NAME } },
      };

      it(`renders No Active Well message if there is rig but not well in appData`, () => {
        const { getByText } = render(withProps(drillingAppProps));

        expect(getByText(getNoActiveWellMessage(RIG_NAME))).toBeInTheDocument();
      });

      testChooseAssetButton(drillingAppProps);
    });

    describe('completion app on general dashboard', () => {
      const completionAppProps = {
        availability: { hasProductSubscription: true, isAppSubscribedForAsset: true },
        AppComponent: {},
        segment: SEGMENTS.COMPLETION,
        layoutEnvironment: { type: 'general' },
        appData: {},
      };

      describe('no frac fleet', () => {
        it(`renders ${APP_MESSAGES.fracFleetWasNotFound.title} messages if there is no frac fleet in appData`, () => {
          const { getByText } = render(withProps(completionAppProps));

          expect(getByText(APP_MESSAGES.fracFleetWasNotFound.title)).toBeInTheDocument();
        });

        testChooseAssetButton(completionAppProps);
      });

      describe('no pad', () => {
        const props = { ...completionAppProps, appData: { fracFleet: {} } };
        it(`renders ${APP_MESSAGES.fracFleetHasNoPad.title} messages if there is no curent pad and padId in settings`, () => {
          const { getByText } = render(withProps(props));

          expect(getByText(APP_MESSAGES.fracFleetHasNoPad.title)).toBeInTheDocument();
        });

        testChooseAssetButton(props);
      });

      describe('no wells', () => {
        const props = { ...completionAppProps, appData: { fracFleet: { current_pad_id: 42 } } };
        it(`renders ${APP_MESSAGES.padHasNoWells.title} messages if there are no wells in appData`, () => {
          const { getByText } = render(withProps(props));

          expect(getByText(APP_MESSAGES.padHasNoWells.title)).toBeInTheDocument();
        });

        it(`renders ${APP_MESSAGES.padHasNoWells.title} messages if there are no wells`, () => {
          const { getByText } = render(
            withProps({
              ...completionAppProps,
              appData: { fracFleet: { current_pad_id: 42 }, wells: [] },
            })
          );

          expect(getByText(APP_MESSAGES.padHasNoWells.title)).toBeInTheDocument();
        });

        testChooseAssetButton(props);
      });
    });

    describe('empty app data', () => {
      const props = {
        availability: { hasProductSubscription: true, isAppSubscribedForAsset: true },
        AppComponent: {},
        appData: {},
      };

      it(`renders ${APP_MESSAGES.noAssetData.title} messages if there are assets`, () => {
        const { getByText } = render(withProps(props));

        expect(getByText(APP_MESSAGES.noAssetData.title)).toBeInTheDocument();
      });

      testChooseAssetButton(props);
    });
  });

  const testRegularApp = (testTitle, additionalProps = {}) =>
    describe(`regular app - ${testTitle}`, () => {
      const COMPONENT_TEST_ID = 'componentTestId';
      const AppComponent = props => (
        <div data-testid={COMPONENT_TEST_ID}>{JSON.stringify(props)}</div>
      );
      const regularAppProps = {
        availability: { hasProductSubscription: true, isAppSubscribedForAsset: true },
        AppComponent,
        appData: {},
        openIntercom: () => {},
        appProps: { appProp1: 'appProp1' },
        app: { ...APP, settings: { rigId: 'rigId' } },
        ...additionalProps,
      };

      it('renders AppComponent without errors', () => {
        const { getByTestId } = render(withProps(regularAppProps));

        expect(getByTestId(COMPONENT_TEST_ID)).toBeInTheDocument();
      });

      it('does not render AppInfoMessage', () => {
        const { queryByTestId } = render(withProps(regularAppProps));

        expect(queryByTestId(APP_INFO_MESSAGE_TEST_ID)).toBeNull();
      });

      // NOTE: props are stringified inside the mocked component
      it('passes all required props to AppComponent', () => {
        const tree = render(withProps(regularAppProps)).baseElement;

        expect(tree).toMatchSnapshot();
      });

      it(`renders ${APP_MESSAGES.internalAppError.title} in case of error`, () => {
        jest.spyOn(console, 'error').mockImplementation(() => {});

        const { getByText } = render(
          withProps({
            ...regularAppProps,
            AppComponent: () => {
              throw new Error('Should be caught by ErrorBoundary');
            },
          })
        );

        expect(getByText(APP_MESSAGES.internalAppError.title)).toBeInTheDocument();
      });
    });

  testRegularApp('with actual appData', { appData: { well: {} } });
  testRegularApp('multirig app', { isMultiRig: true });

  // TODO: add test for openIntercom link
  // TODO: add test for correct image on AppInforPage
});
