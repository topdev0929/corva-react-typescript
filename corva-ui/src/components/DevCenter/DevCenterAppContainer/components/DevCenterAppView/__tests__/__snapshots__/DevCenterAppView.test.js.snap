// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevCenterAppView regular app - multirig app passes all required props to AppComponent 1`] = `
<body>
  <div>
    <div
      data-testid="componentTestId"
    >
      {"app":{"package":{"manifest":{"application":{"name":"My awesome app"}}},"settings":{"rigId":"rigId"}},"rigId":"rigId","appProp1":"appProp1"}
    </div>
  </div>
</body>
`;

exports[`DevCenterAppView regular app - with actual appData passes all required props to AppComponent 1`] = `
<body>
  <div>
    <div
      data-testid="componentTestId"
    >
      {"app":{"package":{"manifest":{"application":{"name":"My awesome app"}}},"settings":{"rigId":"rigId"}},"rigId":"rigId","well":{},"appProp1":"appProp1"}
    </div>
  </div>
</body>
`;
